[DNM] feat(ghostfolio) add ghostfolio

blastik opened the pull request:
**Description**
add [ghostfolio](https://github.com/ghostfolio/ghostfolio) 
submitting in draft so people can add comments

⚒️ Fixes  # <!--(issue)-->

https://github.com/truecharts/charts/issues/7364
https://github.com/truecharts/charts/issues/8117

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [x] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
https://github.com/truecharts/containers/tree/master/mirror/ghostfolio
it was already present
- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

truecharts-admin committed:
chore(deps): update container image tccr.io/truecharts/traefik to v2.10.0 (#8148)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[tccr.io/truecharts/traefik](https://togithub.com/truecharts/containers)
| minor | `2.9.10` -> `2.10.0` |

---

### ⚠ Dependency Lookup Warnings ⚠

Warnings were logged while processing this repo. Please check the
Dependency Dashboard for more information.

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNS41OC4yIiwidXBkYXRlZEluVmVyIjoiMzUuNTguMiJ9-->

---------

Signed-off-by: TrueCharts-Bot <bot@truecharts.org>

Commit:
fix(traefik): revert last container update and update cluster role

Commit:
fix(renovate): don't auto merge minor container updates for enterprise apps to enhance stability

truecharts-admin committed:
chore(deps): update helm general non-major (#8145)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[common](https://togithub.com/truecharts/apps/tree/master/charts/common)
([source](https://togithub.com/truecharts/library-charts)) | minor |
`12.4.21` -> `12.5.1` |
| [mariadb](https://truecharts.org/charts/dependency/mariadb)
([source](https://togithub.com/truecharts/charts)) | patch | `7.0.13` ->
`7.0.15` |
| [mongodb](https://truecharts.org/charts/dependency/mongodb)
([source](https://togithub.com/truecharts/charts)) | patch | `6.0.12` ->
`6.0.14` |
| [redis](https://truecharts.org/charts/dependency/redis)
([source](https://togithub.com/truecharts/charts)) | patch | `6.0.29` ->
`6.0.31` |

---

### ⚠ Dependency Lookup Warnings ⚠

Warnings were logged while processing this repo. Please check the
Dependency Dashboard for more information.

---

### Release Notes

<details>
<summary>truecharts/library-charts</summary>

###
[`v12.5.1`](https://togithub.com/truecharts/library-charts/releases/tag/common-12.5.1)

[Compare
Source](https://togithub.com/truecharts/library-charts/compare/common-12.5.0...common-12.5.1)

Function library for TrueCharts

###
[`v12.5.0`](https://togithub.com/truecharts/library-charts/releases/tag/common-12.5.0)

[Compare
Source](https://togithub.com/truecharts/library-charts/compare/common-12.4.21...common-12.5.0)

Function library for TrueCharts

</details>

<details>
<summary>truecharts/charts</summary>

###
[`v7.0.15`](https://togithub.com/truecharts/charts/releases/tag/mariadb-7.0.15)

[Compare
Source](https://togithub.com/truecharts/charts/compare/mariadb-7.0.14...mariadb-7.0.15)

Fast, reliable, scalable, and easy to use open-source relational
database system.

###
[`v7.0.14`](https://togithub.com/truecharts/charts/releases/tag/mariadb-7.0.14)

[Compare
Source](https://togithub.com/truecharts/charts/compare/mariadb-7.0.13...mariadb-7.0.14)

Fast, reliable, scalable, and easy to use open-source relational
database system.

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "before 10pm on tuesday" in timezone
Europe/Amsterdam, Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config help](https://togithub.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNS41OC4yIiwidXBkYXRlZEluVmVyIjoiMzUuNTguMiJ9-->

---------

Signed-off-by: TrueCharts-Bot <bot@truecharts.org>

truecharts-admin committed:
chore(deps): update helm chart common to 12.6.1 (#8152)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[common](https://togithub.com/truecharts/apps/tree/master/charts/common)
([source](https://togithub.com/truecharts/library-charts)) | minor |
`12.5.1` -> `12.6.1` |

---

### ⚠ Dependency Lookup Warnings ⚠

Warnings were logged while processing this repo. Please check the
Dependency Dashboard for more information.

---

### Release Notes

<details>
<summary>truecharts/library-charts</summary>

###
[`v12.6.1`](https://togithub.com/truecharts/library-charts/releases/tag/common-12.6.1)

[Compare
Source](https://togithub.com/truecharts/library-charts/compare/common-12.5.1...common-12.6.1)

Function library for TrueCharts

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNS41OC4yIiwidXBkYXRlZEluVmVyIjoiMzUuNTguMiJ9-->

truecharts-admin committed:
chore(deps): update helm chart common to 12.6.1 (#8154)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[common](https://togithub.com/truecharts/apps/tree/master/charts/common)
([source](https://togithub.com/truecharts/library-charts)) | minor |
`12.5.1` -> `12.6.1` |

---

### ⚠ Dependency Lookup Warnings ⚠

Warnings were logged while processing this repo. Please check the
Dependency Dashboard for more information.

---

### Release Notes

<details>
<summary>truecharts/library-charts</summary>

###
[`v12.6.1`](https://togithub.com/truecharts/library-charts/releases/tag/common-12.6.1)

[Compare
Source](https://togithub.com/truecharts/library-charts/compare/common-12.5.1...common-12.6.1)

Function library for TrueCharts

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNS41OC4yIiwidXBkYXRlZEluVmVyIjoiMzUuNTguMiJ9-->

blastik committed:
first commit

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/truecharts/charts?pullRequest=8155) <br/>All committers have signed the CLA.

blastik committed:
Merge branch 'master' into feat/add-ghostfolio

xstar97 reviewed (commented):
Chart version should always start as ```0.0.1```

xstar97 commented:
Changelog and readme should just be blank with the first line being a comment.

Those files are pre-generated

truecharts-admin committed:
chore(deps): update helm general non-major (#8153)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[common](https://togithub.com/truecharts/apps/tree/master/charts/common)
([source](https://togithub.com/truecharts/library-charts)) | minor |
`12.5.1` -> `12.6.1` |
| [mariadb](https://truecharts.org/charts/dependency/mariadb)
([source](https://togithub.com/truecharts/charts)) | patch | `7.0.15` ->
`7.0.16` |
| [mongodb](https://truecharts.org/charts/dependency/mongodb)
([source](https://togithub.com/truecharts/charts)) | patch | `6.0.14` ->
`6.0.15` |
| [redis](https://truecharts.org/charts/dependency/redis)
([source](https://togithub.com/truecharts/charts)) | patch | `6.0.31` ->
`6.0.32` |

---

### ⚠ Dependency Lookup Warnings ⚠

Warnings were logged while processing this repo. Please check the
Dependency Dashboard for more information.

---

### Release Notes

<details>
<summary>truecharts/library-charts</summary>

###
[`v12.6.1`](https://togithub.com/truecharts/library-charts/releases/tag/common-12.6.1)

[Compare
Source](https://togithub.com/truecharts/library-charts/compare/common-12.5.1...common-12.6.1)

Function library for TrueCharts

</details>

<details>
<summary>truecharts/charts</summary>

###
[`v7.0.16`](https://togithub.com/truecharts/charts/releases/tag/mariadb-7.0.16)

[Compare
Source](https://togithub.com/truecharts/charts/compare/mariadb-7.0.15...mariadb-7.0.16)

Fast, reliable, scalable, and easy to use open-source relational
database system.

</details>

---

### Configuration

📅 **Schedule**: Branch creation - "before 10pm on tuesday" in timezone
Europe/Amsterdam, Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config help](https://togithub.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNS41OC4yIiwidXBkYXRlZEluVmVyIjoiMzUuNTguMiJ9-->

blastik committed:
fixing precommit

blastik committed:
fix comments

blastik committed:
more improvements

blastik committed:
Merge branch 'master' into feat/add-ghostfolio

blastik committed:
Merge branch 'master' into feat/add-ghostfolio

blastik commented:
@xstar97 any idea why this is failing ? https://github.com/truecharts/charts/actions/runs/4802658444/jobs/8546487667

blastik committed:
more fixes

blastik committed:
Merge branch 'master' into feat/add-ghostfolio

blastik committed:
adding more stuff

Crow-Control commented:
> @xstar97 any idea why this is failing ? https://github.com/truecharts/charts/actions/runs/4802658444/jobs/8546487667

Be sure to carefully read through the tests, they are there to be read:
[Errno 2] No such file or directory: 'charts/incubator/ghostfolio/values.yaml'

We do not have enough maintainers to read it for you on every failure...
Also: Try to tick with copying an existing chart and editing that, instead of DIY'ing, that saves you mistakes like the above.

blastik committed:
yml to yaml

blastik committed:
Merge branch 'master' into feat/add-ghostfolio

blastik commented:
@Ornias1993 how do i fix [this](https://github.com/truecharts/charts/actions/runs/4817875803/jobs/8579918693?pr=8155#step:3:18) and [this](https://github.com/truecharts/charts/actions/runs/4817875803/jobs/8580088113?pr=8155#step:8:48)?

stavros-k commented on charts/incubator/ghostfolio/templates/common.yaml:
```diff
@@ -0,0 +1,53 @@
+  {{- $_ := set .Values.secret "ghostfolio-secrets" $secret -}}
+{{- end -}}
+
+{{/* Append the general configMap volume to the volumes */}}
+{{- define "ghostfolio.configmapVolume" -}}
+enabled: true
+mountPath: /.env
+subPath: .env
+readOnly: true
+type: "configmap"
+objectName: ghostfolio-configfile
+{{- end -}}
+
+{{/* Append the general secret volumes to the volumes */}}
+{{- define "ghostfolio.secretVolumes" -}}
+enabled: true
+mountPath: "/secrets"
+readOnly: true
+type: "secret"
+objectName: ghostfolio-secrets
+items:
+  - key: "ACCESS_TOKEN_SALT"
+    path: ACCESS_TOKEN_SALT
+  - key: "JWT_SECRET_KEY"
+    path: JWT_SECRET_KEY
+  - key: "DATABASE_URL"
+    path: DATABASE_URL
+  - key: "REDIS_PASSWORD"
+    path: REDIS_PASSWORD
+{{- end -}}
```
There is no need for those (persistence's) to be in the common.yaml.
There is no conditionals. Please move them to values.yaml

stavros-k commented on charts/incubator/ghostfolio/values.yaml:
```diff
@@ -0,0 +1,48 @@
+    runAsUser: 0
+    runAsGroup: 0
+
+alpha_vantage_api_key: ""
```
Please move app specific config under a dict like
```yaml
ghostfolio:
  somekey: somevalue
```

stavros-k commented on charts/incubator/ghostfolio/templates/common.yaml:
```diff
@@ -0,0 +1,53 @@
+readOnly: true
+type: "secret"
+objectName: ghostfolio-secrets
+items:
+  - key: "ACCESS_TOKEN_SALT"
+    path: ACCESS_TOKEN_SALT
+  - key: "JWT_SECRET_KEY"
+    path: JWT_SECRET_KEY
+  - key: "DATABASE_URL"
+    path: DATABASE_URL
+  - key: "REDIS_PASSWORD"
+    path: REDIS_PASSWORD
```
Are those intentionally mounted as files?

stavros-k commented on charts/incubator/ghostfolio/templates/_secrets.tpl:
```diff
@@ -0,0 +1,22 @@
+    {{ $accesstokensalt = (index .data "ACCESS_TOKEN_SALT") | b64dec }}
+    {{ $jwtsecret = (index .data "JWT_SECRET_KEY") | b64dec }}
+
+  {{ end }}
```
```suggestion

  {{- with (lookup "v1" "Secret" .Release.Namespace $fetchname) -}}
    {{/* Get previous values and decode */}}
    {{- $accesstokensalt = (index .data "ACCESS_TOKEN_SALT") | b64dec -}}
    {{- $jwtsecret = (index .data "JWT_SECRET_KEY") | b64dec -}}
  {{- end }}
  
enabled: true
data:  
```

stavros-k committed:
Update charts/incubator/ghostfolio/templates/_secrets.tpl

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k commented on charts/incubator/ghostfolio/templates/_secrets.tpl:
```diff
@@ -0,0 +1,22 @@
+data:  
+  ACCESS_TOKEN_SALT: {{ $accesstokensalt }}
+  JWT_SECRET_KEY: {{ $jwtsecret }}
+  DATABASE_URL: "postgresql://{{ default "ghostfolio" .Values.cnpg.main.creds.user }}:{{ .Values.cnpg.main.creds.password }}@{{ .Values.cnpg.main.creds.host }}:{{ default 5432 .Values.cnpg.main.creds.port }}"
```
```suggestion
  DATABASE_URL: {{ printf "postgresql://%v:%v@%v:5432" .Values.cnpg.main.creds.user .Values.cnpg.main.creds.password .Values.cnpg.main.creds.host }}
```

stavros-k committed:
Update charts/incubator/ghostfolio/templates/_secrets.tpl

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k commented on charts/incubator/ghostfolio/values.yaml:
```diff
@@ -0,0 +1,48 @@
+
+workload:
+  main:
+    replicas: 1
```
this is the default
```suggestion
```

stavros-k committed:
Update charts/incubator/ghostfolio/values.yaml

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k commented on charts/incubator/ghostfolio/questions.yml:
```diff
@@ -0,0 +1,22 @@
+portals:
+  open:
+# Include{portalLink}
+questions:
+  # Include{global}
+  # Include{workload}
+  # Include{workloadDeployment}
+  # Include{replicas1}
+  # Include{podSpec}
+  # Include{containerMain}
+  # Include{containerBasic}
+  # Include{containerAdvanced}
+  # Include{containerConfig}
+  # Include{podOptions}
+  - variable: alpha_vantage_api_key
+    group: "App Configuration"
+    label: "alpha_vantage_api_key"
+    description: "https://www.alphavantage.co/ API key"
+    schema:
+      type: string
+      default: ""
```
This lacks 60% of the UI

stavros-k commented on charts/incubator/ghostfolio/templates/_configmap.tpl:
```diff
@@ -0,0 +1,27 @@
+
+    # CACHE
+    REDIS_HOST={{ .Values.redis.creds.plain }}
+    {{- with $redis := .Values.redisProvider }}
```
What is this `with`? There is no `redisProvider` in values.
Even if it was, it won't be optional

stavros-k commented on charts/incubator/ghostfolio/templates/_configmap.tpl:
```diff
@@ -0,0 +1,27 @@
+    {{- end }}
+
+    # POSTGRES
+    POSTGRES_DB={{ default "ghostfolio" .Values.cnpg.main.creds.database }}
```
```suggestion
    POSTGRES_DB={{ .Values.cnpg.main.creds.database }}
```
Having default values for deps connection details will only create confusion.
if the cpng value missing, means something is wrong anyway. No need to fallback to sometihng else

stavros-k reviewed (changes requested):
.

Crow-Control commented:
closed for being stale.

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
